### DISCUSSION TITLE
Godbolt viewer

### DISCUSSION THREAD

**Discussion by sudakshinadutta** [intel/llvm#6501] [Q&A]
Greetings from India. Please help me to understand a few things if possible.

Please go to the link https://godbolt.org/z/z375aK4KK where the LLVM Opt pipeline viewer is shown. I am unable to find the output shown in the windows. Some passes are listed in a window against which two vertical windows are displayed. As the passes change, the outputs of the windows change.  Can you kindly help me in understanding the meaning of the output shown in the marked windows of the attached file ? More precisely, I need to know what do the contents of the windows imply ? This is needed for the research work I am doing currently.
![godbolt](https://user-images.githubusercontent.com/50015559/181921407-f107299a-b926-4e20-9f7e-5fa803d3cfc4.png)

**Comment by AlexeySachkov** [comment]
Hi @sudakshinadutta, 

This question is probably better be asked in https://github.com/llvm/llvm-project, because it doesn't seem to have anything to do with Intel oneAPI DPC++ compiler, which is hosted in this repo.

> Some passes are listed in a window against which two vertical windows are displayed. As the passes change, the outputs of the windows change. Can you kindly help me in understanding the meaning of the output shown in the marked windows of the attached file ? More precisely, I need to know what do the contents of the windows imply ?

As it is stated at the top of each window, they contain machine code for a certain function from your input file. My understanding is that such split view probably shows before/after, i.e. you select a pass and you see its input and its output, i.e. you can explore changes the pass made.

**Reply by aelovikov-intel** [reply]
And https://discourse.llvm.org/ might be an even better place to ask it.
